code/core/write_sdl_ntuple.cc

```diff
@@ -782,6 +843,195 @@ void setGnnNtupleMiniDoublet(SDL::Event* event, unsigned int MD)
+        Pictorial representation of a T3
+       
+        inner tracker  outer tracker
+        -------------  --------------
+                       01    23    45   (anchor hit of a minidoublet is always the first of the pair)
+         (none)        oo -- oo -- oo   T3
+    */
+
+    // Get relevant information
+    SDL::hits& hitsInGPU = (*event->getHits());
+    SDL::modules& modulesInGPU = (*event->getFullModules());
+    SDL::triplets& tripletsInGPU = (*event->getTriplets());
+
+    // Hits
+    std::vector<unsigned int> hits = getHitsFromT3(event, T3);
+    unsigned int hit0 = hits[0];
+    unsigned int hit2 = hits[2];
+    unsigned int hit4 = hits[4];
+
+    // Hit locations
+    const float hit0_x = hitsInGPU.xs[hit0];
+    const float hit0_y = hitsInGPU.ys[hit0];
+    const float hit0_z = hitsInGPU.zs[hit0];
+    const float hit0_r = sqrt(hit0_x * hit0_x + hit0_y * hit0_y);
+    const float hit2_x = hitsInGPU.xs[hit2];
+    const float hit2_y = hitsInGPU.ys[hit2];
+    const float hit2_z = hitsInGPU.zs[hit2];
+    const float hit2_r = sqrt(hit2_x * hit2_x + hit2_y * hit2_y);
+    const float hit4_x = hitsInGPU.xs[hit4];
+    const float hit4_y = hitsInGPU.ys[hit4];
+    const float hit4_z = hitsInGPU.zs[hit4];
+    const float hit4_r = sqrt(hit4_x * hit4_x + hit4_y * hit4_y);
+    ana.tx->pushbackToBranch<float>("t3_0_r", hit0_r);
+    ana.tx->pushbackToBranch<float>("t3_0_x", hit0_x);
+    ana.tx->pushbackToBranch<float>("t3_0_y", hit0_y);
+    ana.tx->pushbackToBranch<float>("t3_0_z", hit0_z);
+    ana.tx->pushbackToBranch<float>("t3_2_r", hit2_r);
+    ana.tx->pushbackToBranch<float>("t3_2_x", hit2_x);
+    ana.tx->pushbackToBranch<float>("t3_2_y", hit2_y);
+    ana.tx->pushbackToBranch<float>("t3_2_z", hit2_z);
+    ana.tx->pushbackToBranch<float>("t3_4_r", hit4_r);
+    ana.tx->pushbackToBranch<float>("t3_4_x", hit4_x);
+    ana.tx->pushbackToBranch<float>("t3_4_y", hit4_y);
+    ana.tx->pushbackToBranch<float>("t3_4_z", hit4_z);
+
+    /* Sigmas for chi2 calculation 
+     * (stolen from SDL::computeSigmasForRegression and SDL::computeRadiusUsingRegressionk) */
+    std::vector<float> sigmas;
+    float inv1 = 0.01f/0.009f;
+    float inv2 = 0.15f/0.009f;
+    // float inv3 = 2.4f/0.009f; // not used
+    for (auto hit : {hit0, hit2, hit4})
+    {
+        // Get module info
+        unsigned int module = hitsInGPU.moduleIndices[hit];
+        SDL::ModuleType module_type = modulesInGPU.moduleType[module];
+        short module_subdet = modulesInGPU.subdets[module];
+        short module_side = modulesInGPU.sides[module];
+        float module_drdz = modulesInGPU.drdzs[module];
+        float module_slope = modulesInGPU.slopes[module];
+        // Get deltas for sigma calculation
+        float delta1, delta2;
+        bool is_flat;
+        // Category 1: barrel PS flat
+        if (module_subdet == SDL::Barrel and module_type == SDL::PS and module_side == SDL::Center)
+        {
+            delta1 = inv1;//1.1111f;//0.01;
```

Comment:
Commented out numbers should be deleted? The first one seems to be the result of `inv1` (so why have it re-written), while the second one is random - at least I can't understand where it comes from...

This comment applies to similar instances below.